src_c/rwobject.c

```diff
@@ -709,11 +709,9 @@ _rwops_from_pystr(PyObject *obj)
             strcpy(extension, ext);
         }
         rw->hidden.unknown.data1 = (void *)extension;
-        Py_DECREF(oencoded);
         return rw;
     }
 
-    Py_DECREF(oencoded);
```

Comment:
the changes in this file should be reverted. This was a recent fix I made to fix an issue with DECREFs. Basically, `oencoded` should not be freed while `encoded` is still being used

Reply:
right i missed the strrchr(encoded, '.');